backend/src/packages/orders/order.service.ts

```diff
@@ -0,0 +1,168 @@
+import { type UserEntityObjectWithGroupT } from 'shared/build/index.js';
+
+import { type IService } from '~/libs/interfaces/interfaces.js';
+
+import { OrderStatus } from './libs/enums/enums.js';
+import { OrderNotFound } from './libs/exceptions/order-not-found.js';
+import {
+  type OrderCreateRequestDto,
+  type OrderCreateResponseDto,
+  type OrderEntity as OrderEntityT,
+  type OrderUpdateResponseDto,
+} from './libs/types/types.js';
+import { OrderEntity } from './order.entity.js';
+import { type OrderRepository } from './order.repository.js';
+
+class OrderService implements Omit<IService, 'delete' | 'findById'> {
+  private orderRepository: OrderRepository;
+
+  public constructor(orderRepository: OrderRepository) {
+    this.orderRepository = orderRepository;
+  }
+
+  public async create(
+    payload: OrderCreateRequestDto & {
+      userId: number | null;
+      businessId: number;
+    },
+  ): Promise<OrderCreateResponseDto> {
+    const {
+      scheduledTime,
+      carsQty,
+      startPoint,
+      endPoint,
+      customerName,
+      customerPhone,
+      driverId,
+      userId,
+      businessId,
+    } = payload;
+    const order = await this.orderRepository.create(
+      OrderEntity.initializeNew({
+        price: 100, //Mock, get price from driver->user->truck and calculated distance
+        scheduledTime,
+        carsQty,
+        startPoint,
+        endPoint,
+        status: OrderStatus.PENDING,
+        userId,
+        businessId,
+        driverId,
+        customerName,
+        customerPhone,
+      }),
+    );
+
+    return order.toObject();
+  }
+
+  public async findById(
+    id: OrderEntityT['id'],
+    user: UserEntityObjectWithGroupT | null | undefined,
+  ): Promise<OrderEntityT | null> {
+    const foundOrder = await this.orderRepository.findById(id);
+
+    if (!foundOrder) {
+      throw new OrderNotFound();
+    }
+    const result = foundOrder.toObject();
+
+    if (user) {
+      this.checkIfOrderBelongsToUser(result, user);
+    }
+
+    return result;
+  }
+
+  public async update(parameters: {
+    id: OrderEntityT['id'];
+    payload: Partial<OrderEntityT>;
+    driverId: number | undefined;
+  }): Promise<OrderUpdateResponseDto> {
+    const foundOrder = await this.orderRepository.findById(parameters.id);
+
+    if (parameters.driverId && foundOrder) {
+      this.checkIfOrderBelongsToDriver(
+        foundOrder.toObject(),
+        parameters.driverId,
+      );
+    }
+
+    const updatedOrder = await this.orderRepository.update(parameters);
+
+    if (!updatedOrder) {
+      throw new OrderNotFound();
+    }
+
+    return updatedOrder.toObject();
+  }
+
+  public async find({
+    userId,
+    businessId,
+    driverId,
+    currentUserBusinessId,
+    currentUserId,
+  }: {
+    userId: string | undefined;
+    businessId: string | undefined;
+    driverId: string | undefined;
+    currentUserBusinessId: number | undefined;
+    currentUserId: number | null;
```

Comment:
if you want to write such generic find then use ?
and again, why ids are strings ?
for me, I don't like this idea of almost all undefined fields. I prefer more approach with separate methods for finding, but each for special purpose.

```suggestion
    userId?: string;
    businessId?: string;
    driverId?: string;
    currentUserBusinessId:? number | undefined;
    currentUserId: number | null;
```

Reply:
ok, I'll use '?', also I'll change type to 'number', as joy worked with query strings interpreting string like numbers, and using string type was not good idea. I suppose that at some point we will use filters to find all orders made by driver that belongs to business, all orders  made by user from particular business and so on. I can suggest to make this method private and use it from other methods. What do you think? But If you insist I l'll make separate methods

Reply:
I'm also more towards having separate findById, findByUserId, findBySomething inside services.
Although, I'd consider to have a more versatile "findByAnything"-like method as part of your orderRepository and use it in every findBy... of your service.